src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -3177,12 +3177,19 @@ public void GetContainerProperties()
         /// <summary>
         /// Dynamically load the Host Compute interop assemblies and return useful types.
         /// </summary>
-        /// <param name="computeSystemPropertiesType">The Microsoft.HyperV.Schema.Compute.System.Properties type.</param>
+        /// <param name="computeSystemPropertiesType">The HCS.Compute.System.Properties type.</param>
         /// <param name="hostComputeInteropType">The Microsoft.HostCompute.Interop.HostComputeInterop type.</param>
         private static void GetHostComputeInteropTypes(out Type computeSystemPropertiesType, out Type hostComputeInteropType)
         {
             Assembly schemaAssembly = Assembly.Load(new AssemblyName("Microsoft.HyperV.Schema, Version=10.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"));
-            computeSystemPropertiesType = schemaAssembly.GetType("Microsoft.HyperV.Schema.Compute.System.Properties");
+
+            // The type name was changed in newer version of Windows so we check for new one first,
+            // then fallback to previous type name to support older versions of Windows
+            computeSystemPropertiesType = schemaAssembly.GetType("HCS.Compute.System.Properties");
+            if (computeSystemPropertiesType == null)
```

Comment:
Now that we know namespace, type properties and fields can and will change on us, I think we need to check for null values and provide a better message than "Null Reference Exception".  In the longer term we should work with the Hyper-V team to have a less fragile way to get this information.  I agree with @iSazonov, that these APIs should be public and supported.

Reply:
I'll update this to be more robust